geps/gep-1651.md

```diff
@@ -0,0 +1,154 @@
+# GEP-1651: GatewayAddress Scope
+
+* Issue: [#1651](https://github.com/kubernetes-sigs/gateway-api/issues/1651)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## TLDR
+
+Allow users to configure a GatewayAddress so that it is only accessible within
+a specific scope (ie. public/cluster local)
+
+## Goals
+
+- Cluster local gateways can be created in a declarative way `ClusterLocal` and
+  are only accessible within a single cluster
+- Define a `Public` scope for addresses that are accessible to the internet
```

Comment:
If I use a LoadBalancer Service and set the annotations to be internal-facing, e.g. only within my AWS VPC, is this considered `public` or `ClusterLocal`? According to the current definitions, it doesn't comply with either.

Reply:
GatewayClass [parametersRef](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.ParametersReference) was defined to meet this type of use case. Can you help me understand why parametersRef does not meet your needs?

Reply:
> If I use a LoadBalancer Service and set the annotations to be internal-facing, e.g. only within my AWS VPC, is this considered public or ClusterLocal? According to the current definitions, it doesn't comply with either.

There was a discussion around this in this comment - https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1441013384

For now I left this open and you could use a vendored prefix scope. Subsequent GEPs could define additional standard scopes like Rob suggested.

Reply:
flushed out a section about gateway class as an alternative